### ISSUE TITLE
Auto-discovery fails block every operation

### ISSUE THREAD

**Issue by akamensky** [wrouesnel/postgres_exporter#383]
Details:

- Exporter version 0.8.0 (downloaded from published releases)
- Postgres 10.9
- Envrionment Centos 7.5 64-bit

Steps:

1. Run Postgres with default ident authentication
2. Prepare queries file locally same as the one in this repo
2. Run exporter locally (as supposed with ident) with auto discovery enabled as postgres user (superuser) and connecting to unix socket using `DATA_SOURCE_NAME="user=postgres host=/var/run/postgresql database=postgres sslmode=disable"` and using local queries

Expected outcome:

Auto-discovery finds all databases and can connect to them, every refresh of metrics page from exporter is "instantaneous"

Actual outcome:

Startup always delayed by `num_of_dbs * 6` seconds. Every refresh of metrics page takes exactly  `num_of_dbs * 6` seconds. exporter is FAILING to connect to all discovered DBs.

NOTE:

I already gave it unrestricted superuser access and it is still FAILING to connect to DBs? No, I am not going to use md5 auth over TCP sockets, I want to use ident running locally from restricted user, however appears ident even when using superuser is not working properly.

**Comment by frittentheke** [comment]
@akamensky  Is this the same issue as https://github.com/wrouesnel/postgres_exporter/issues/293 ?

**Comment by akamensky** [comment]
@frittentheke not exactly, this is about the timing it needs to fail. 6 seconds per each existing database is a bit too much, we are trying to pull data every 10 seconds, while EVERY request takes about 3 minutes because of number of different databases this instance contains.

**Comment by sholz** [comment]
I am facing similar issues. With auto-discovery enabled everything is very slow. 

I am seeing log messages which I think are indicating that prometheus closes connections already while results are still being calculated:

```
http: multiple response.WriteHeader calls
```

Also liveness issues on k8s are occurring when auto-discovery is enabled.

Without auto-discovery everything works fine.

**Comment by sysadmind** [comment]
v0.19.0 added a bounded scrape duration (#1229) so a down database no longer hangs the metrics endpoint indefinitely. Closing as resolved; please reopen a new issue if you still see this on a current release.

**Comment by sysadmind** [comment]
--auto-discover-databases was also deprecated in v0.13.0

**Comment by akamensky** [comment]
Would be great news if it didn't take 6 years and we didn't already move on from Prometheus altogether.

**Comment by sysadmind** [comment]
Thanks for the feedback. We are currently working to clean up the backlog that has grown too large.
